source/material_model/thermal_conductivity/Hofmeister1999.cc

```diff
@@ -0,0 +1,224 @@
+        // mineral composition [(Mg1.8Fe0.2)SiO4]
+        constexpr int RingwooDry_Index = 2;
+        const double RingwooDry_Hof99_LatTC_Room =  7.7;        // lattice thermal conductivity at room temperature (Lambda_Room)
+        const double RingwooDry_Hof99_LatTC_TExp =  0.3;        // temperature exponent (N_Texp)
+        const double RingwooDry_Hof99_LatTC_GruP =  1.25;       // Grueneisen parameter (Gamma)
+        const double RingwooDry_Hof99_LatTC_Alph =  2.0535e-5;  // thermal expansion coefficient (Alpha)
+        const double RingwooDry_Hof99_LatTC_Bulk =  183;        // bulk modulus (K0)
+        const double RingwooDry_Hof99_LatTC_PDev =  5.2;        // pressure derivative of bulk modulus (K_prime)
+        const double RingwooDry_Hof99_RadTC_CoA0 =  1.7530e-2;  // coefficient for radiative thermal conductivity T^0 (A0)
+        const double RingwooDry_Hof99_RadTC_CoB1 = -1.0365e-4;  // coefficient for radiative thermal conductivity T^1 (B1)
+        const double RingwooDry_Hof99_RadTC_CoC2 =  2.2451e-7;  // coefficient for radiative thermal conductivity T^2 (C2)
+        const double RingwooDry_Hof99_RadTC_CoD3 = -3.4071e-11; // coefficient for radiative thermal conductivity T^3 (D3)
+
+        unsigned int MineralPar_Index = RingwooDry_Index+1; // Number of minerals
+
+        // Define room temperature [K] 
+        const double T_room = 298.15; 
+        
+        const unsigned int n_points = in.n_evaluation_points();
+
+        for (unsigned int i = 0; i < n_points; ++i) 
+        {
+
+         // Preallocate a vector for storing thermal conductivities of minerals
+         std::vector<double> Hof99_Minerals_LatTcond(MineralPar_Index, 0.0); // Lattice thermal conductivity
+         std::vector<double> Hof99_Minerals_RadTcond(MineralPar_Index, 0.0); // Radiative thermal conductivity
+         std::vector<double> Hof99_Minerals_TotTcond(MineralPar_Index, 0.0); // Total thermal conductivity
+         // Preallocate a matrix for storing thermal conductivities of minerals
+         std::vector<std::vector<double>> Hof99_All_Minerals_TConds(MineralPar_Index, std::vector<double>(3, 0.0));
+
+         // Convert pressure unit from [Pa] to [GPa]
+         double P_GPa = in.pressure[i]/1e9;
+         // Take the temperature field of the model [K]
+         double T_mod = in.temperature[i];
+         // Take the mineral fraction of the model
+         double min_frac = in.composition[0][i];
+ 
+         unsigned int mID = in.Mineral_ID;
+
+         switch (mID) // Compute the lattice, radiative and total thermal conductivities of the given mineral
+         {
+           case OlivineDry_Index: // Dry Olivine
+           { 
+             double OlivineDry_Hof99_LatTCon = Compute_Lat_TCond_Hofmeister1999(
+             OlivineDry_Hof99_LatTC_Room, OlivineDry_Hof99_LatTC_TExp, OlivineDry_Hof99_LatTC_GruP, OlivineDry_Hof99_LatTC_Alph,
+             OlivineDry_Hof99_LatTC_Bulk, OlivineDry_Hof99_LatTC_PDev, T_room, T_mod, P_GPa);   
```

Comment:
We usually write long parameter lists like the following:
```suggestion
double OlivineDry_Hof99_LatTCon = Compute_Lat_TCond_Hofmeister1999(OlivineDry_Hof99_LatTC_Room,
                                 OlivineDry_Hof99_LatTC_TExp,
                                 OlivineDry_Hof99_LatTC_GruP,
                                 OlivineDry_Hof99_LatTC_Alph,
                                 OlivineDry_Hof99_LatTC_Bulk,
                                 OlivineDry_Hof99_LatTC_PDev,
                                 T_room,
                                 T_mod,
                                 P_GPa);   
```

(i.e. one line per parameter and all parameters indented to the opening parenthesis of the function call). Just a stylistic question, but we try to be consistent.